scripts/core/EXP-COMMAND-BUFFER.rst

```diff
@@ -89,6 +94,12 @@ However, they differ in that they take a command-buffer handle instead of a
 queue handle, and the dependencies and return parameters are sync-points instead
 of event handles.
 
+The entry-point for appending a kernel launch command also returns an optional
+handle to the command being appended. Returning this handle extends the
+lifetime of the parent command-buffer by incrementing its reference count, and
+the parent command-buffer's reference count is decremented again when the command
+handle is destroyed.
```

Comment:
To me this seems a bit counterintuitive, I would've thought that the command-buffer would extend the lifetime of the command handles in case it needed them internally. They can't both extend each other's lifetimes, and I'm not sure this matters that much, but was there a specific reason this way was chosen?

Reply:
There's nothing really specified about reference counting in the UR spec, so this behaviour is based on what adapter implementations are doing for kernel/program & event refcounting. Before I was looking at the CUDA adapter which doesn't do anything fancy with refcounting, but now looking into Level Zero it has internal & external event refcounting implemented (presumably for this reason).

I'll try use that as inspiration to do something similar for command handles, because I agree it makes sense.

Reply:
I've pushed a commit which implements command-buffers and command-handles with an internal reference count so that they can extend each others lifetimes. I did however also remove this spec wording, since this is turning into more of an implementation detail and reference counting isn't really talked about elsewhere.